### PR TITLE
Cleanup redundancy in configuration file

### PR DESCRIPTION
**Pull request by ashmaroli**
- Remove config keys that *do not change* defaults set by Jekyll
- Improve overall readability:
  - Move colons nearer to the "value" so that they do not appear disconnected from the key
    (trailing whitespace to the key have no effect)
  - Group *YAML lists* together

### CODE REVIEW

**Comment by XhmikosR** [comment]
I don't like the spaces before `:`. Also, explicitly setting some defaults is done by choice, because I'd rather be explicit than having to track all defaults with the GitHub Jekyll version.

**Comment by XhmikosR** [comment]
Also note that our base config is upstream https://github.com/twbs/bootstrap/blob/v4-dev/_config.yml

For the rest, we follow that for consistency reasons.

File: _config.yml [hunk 1]
```diff
@@ -1,30 +1,30 @@
 # Setup
-title             : Bootstrap Blog
-description       : Official blog for the Bootstrap framework.
-url               : "https://blog.getbootstrap.com"
-future            : true
-permalink         : pretty
+title:              "Bootstrap Blog"
```
Comment (ashmaroli) [inline _config.yml]: The quotes are not necessary actually. Is there a reason why you're bringing them back..?

Comment (XhmikosR) [reply _config.yml]: Yeah, I'm going with the safe way http://www.yamllint.com/

Comment (ashmaroli) [reply _config.yml]: Okay. Fair enough :smiley:

Comment (XhmikosR) [reply _config.yml]: I just need a way to (semi) automate the validity of the file, that's all :)

**Comment by ashmaroli** [comment]
> But we need to take into consideration the mismatch in the local Jekyll version and the GitHub one

That the reason a `Gemfile.lock` exists.
Github Pages is currently locked to `Jekyll 3.7.x`. And there wasn't any "major" changes in v3.8 either.
Besides, Jekyll 3.x is no longer under active development. Any "change" will only affect v4.0

File: _config.yml (hunk 1 above)
Comment (ashmaroli) [reply _config.yml]: If formatting and comments were not an issue, automation could be carried out by a Rake task or a standalone Ruby script. :smiley: 
```ruby
require "safe_yaml/load"

config_file = "./_config.yml"
config = SafeYAML.load_file(config_file)
File.write(config_file, YAML.dump(config))
```

**Comment by XhmikosR** [comment]
Sorry, GitHub pages does not respect the lockfile and I have no time or interest to follow this too.

I'd rather be explicit.

**Comment by ashmaroli** [comment]
Is there anything in this PR you'd like to use on the base-branch..?

**Comment by XhmikosR** [comment]
@ashmaroli: I'm still not sure if we should go with this yet. I think it probably is fine to merge it as long as we make the same changes in bootstrap and bootstrap-expo too later.

**Comment by XhmikosR** [comment]
BTW this can't be merged, we need a rebase not a merge.

**Comment by ashmaroli** [comment]
> we need a rebase not a merge

Ohkay no probs, I though it didn't matter since you squash-merge all PRs anyway.
I'll rebase and force-push shortly.

**Pull request merged**